azavea/raster-vision
# Run experiments from CLI

## Overview

Implements the CLI for `rastervision`. This is the CLI that can be run from experiment files that define an `ExperimentSet` (renamed from `ExperimentSuite`), and define:

```
if __name__ ==  '__main__':
    rv.main()
```

This is the CLI that should be set up as the `rastervision` command with setuptools for the pip installable Raster Vision as described in the click documentation: http://click.pocoo.org/5/setuptools/

This is the base of the CLI, which should be improved  and commands and features added in future PRs.

## Depends on:
- #383 
- #385 

## Testing Instructions

You can  run the experiments contained in the integration tests by the following commands  inside the docker container, from the `/opt/src` directory:

```shell
> python integration_tests/object_detection_tests/experiment.py run local -a tmp_dir `pwd`/tmp_test chip
> python -m integration_tests.object_detection_tests.experiment run local -a tmp_dir `pwd`/tmp_test chip
> python -m rastervision run local -e integration_tests.object_detection_tests.experiment -a tmp_dir `pwd`/tmp_test chip
```

Notice we have to use the`-a KEY VALUE` option to pass arguments into the experiment methods, since the integration tests experiment methods  take a tmp_dir non-default parameter. This will create a `tmp_test` directory in your /opt/src directory.

You can also list experiments with 
```shell
> python -m integration_tests.object_detection_tests.experiment ls -a tmp_dir `pwd`/tmp_test
```
